### PR TITLE
Additional ents

### PR DESCRIPTION
**Pull request by dsvensson**
This brings along the complete set of entities used in the RCTF maps by infiniti. All of them widely available and used in the SP mapping community, nothing too fringe. Should be lots of room in mvdsv for a more bloated edict now with the lean bsp loading in place.

Intend to put together a FGD (TrenchBroom description file) in a separate repository to document these entities together with the existing ktx custom entities so it's easier for mappers to target ktx.

TL;DR:
* ambient_general - like the other ambient_* but with custom wav.
* func_bob - smooth bobbing model, can't really be done with func_train.
* func_laser - a forcefield door-like entity with pulsating alpha value that can damage player on touch.
* hiprot - func_rotate etc, from Scourge of Armagon mission pack, hated but useful.
* trigger_heal - a trigger that heals players.

### CODE REVIEW

**Comment by tcsabina** [comment]
Nano,

What part of this changeset would you like to see in qwprot?

**Comment by tcsabina** [comment]
Ignore previous comment, please. It made no sense ;)

**Comment by dsvensson** [comment]
Here are the RCTF maps, https://bit.ly/rctf_b2 .. rctf3_b2 currently doesn't load in ezQ due to a bug, and func_laser on rctf1_b2 without mvdsv/ezq alpha is a bit boring, but could verify all of this in fteqw-sv + fteqw. Waiting for feedback from infinity as well for this branch but it looks like it's working as intended. They are all very mechanical line-by-line ports from QuakeC to the KTX helper functions.

**Comment by tcsabina** [comment]
@inf1niti 
let us know when ready to merge.

**Comment by inf1niti** [comment]
Finally tested this 😅, generally looks good.

Couple of small things:
1. I wasn't able to test the `trigger_heal`, but I assume it works.
2. sounds ref'd from `ambient_general` and other ents (ie. `func_laser`) were not playing for me.  Got these warnings in the console (maybe I've got my sounds in the wrong folder or something?) 
3. Was getting console-ddos'd with alpha-related error spam when running rctf1.. This is due to entities with alpha transparency being embedded in the map, and not directly related to this PR **but worth noting anyway, that any map that includes the alpha property on a bmodel will spam players unendingly with the following message:

**Comment by dsvensson** [comment]
The alpha spam is only for func_laser as it changes alpha over time. The syscall could be updated to log once per map in a separate commit.

The server could support alpha as well :) should finish that.

**Comment by dsvensson** [comment]
@inf1niti The ambient sounds in rctf package lack cue markers - they do have other types of markers that varies from file to file, but not the `ltxt`+`mark` that ezq looks for. fte is more relaxed and only looks for `mark`. hexdump any of them and compare with idSoftware ambience/*.wav, more info on https://quakewiki.org/wiki/ambientsound ... do they loop correctly in other engines?

**Comment by dsvensson** [comment]
The wav files can be modified with python or something to adhere to the ltxt+mark way, and a future improvement to ezq would be to parse additional marker types in wav files to make it easier to work with audio tools from this century.

A workaround to verify that ambient_general does what it says it should is to add this to S_LoadSound:

```c
	if (loopstart == -1 && strncmp(s->name, "ambience/", 9) == 0) {
		loopstart = 0;
	}
```

at the end before resample.

**Review by tcsabina — CHANGES_REQUESTED** [review]

File: include/progs.h [hunk 1]
```diff
@@ -1200,6 +1200,33 @@ typedef struct gedict_s
...
+
+// { ambient_general
+	float volume;
+// }
+
+// { trigger_heal
+	float healmax;
+	float healtimer;
+// }
 } gedict_t;
```
[the remark this answers is not in the document]
Comment (dsvensson) [reply include/progs.h:1230]: Don't expect so. It's all allocated up front during mod init and cmodel.c used to need a ton of waste mem to load some of the newer maps. I fixed that so there should be lots of headroom now. Would be nice with some bitflag indirection or something to wipe out that struct, but think it's fine. In practice I could have reused a lot more as they are per ent and most aren't used in the same ent, but readability would suffer by using weird names for things. Have tried to reuse as many already declared members where the name was close enough.

File: include/q_shared.h
[the remark this answers is not in the document]
Comment (dsvensson) [reply include/q_shared.h:1]: Yes. And also minor task of moving mvdsv into ezq repo :)

**Comment by dsvensson** [comment]
The comments wrt ambient_general are resolved by passing the wav files through this[[1]](https://github.com/qw-ctf/ambiencefix) utility that takes a `timestamp` in float seconds, `input` file, and `output` file argument and rewrites the RIFF header to fit ezquake - and less strict engines, 

1. https://github.com/qw-ctf/ambiencefix

**Review by tcsabina — APPROVED** [review]
